perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/pwdmgr/EinfraPasswordManagerModule.java

```diff
@@ -67,6 +68,24 @@ public EinfraPasswordManagerModule() {
 
 	}
 
+	@Override
+	public String handleSponsorship(PerunSession sess, SponsoredUserData userData) {
+		if (StringUtils.isBlank(userData.getPassword())) {
+			throw new InternalErrorException("Einfra namespace requires a password.");
+		}
+		if (StringUtils.isBlank(userData.getNamespace())) {
+			throw new InternalErrorException("Einfra requires a login.");
+		}
+
+		try {
+			reservePassword(sess, userData.getLogin(), userData.getPassword());
```

Comment:
I still think, that you should throw those exceptions outside and not wrap them.

Reply:
Done.